packages/react/context-menu/src/ContextMenu.tsx

```diff
@@ -14,14 +14,13 @@ import type { Point, MeasurableElement } from '@radix-ui/utils';
 const CONTEXT_MENU_NAME = 'ContextMenu';
 
 type ContextMenuContextValue = {
-  open: boolean;
-  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
   anchorPointRef: React.MutableRefObject<Point>;
   anchorRef: React.MutableRefObject<MeasurableElement | null>;
+  open: boolean;
+  onOpenChange(open: boolean): void;
```

Comment:
Again, not sure how I feel here, even the change of signature, we're putting a setter in it, what if I actually want to use the `prevState => newState` syntax? I'm kinda confused why we're changing all these… I'm probably not seeing something.